### DISCUSSION TITLE
How to get typechecking of supabase functions to work?

### DISCUSSION THREAD

**Discussion by carcinocron** [supabase/supabase#17184] [Questions]
if I run:
```bash
# defined as: "typecheck": "tsc -p ./tsconfig.json",
npm run typecheck
```
the result is:
```
supabase/functions/hello-world/index.ts:1:43 - error TS2307: Cannot find module 'https://deno.land/std@0.199.0/http/server.ts' or its corresponding type declarations.

1 import { serve, type ServerRequest } from 'https://deno.land/std@0.199.0/http/server.ts'
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 errors in 2 files.

Errors  Files
     1  supabase/functions/hello-world/index.ts:1
```
The rest of my repo is solidjs from a starter if that matters.

**Comment by justinekatebe** [comment]
What worked for me was in VSCode pressing shift + command + p (on MacOS), then searching for "Deno Initialize workspace configuration" then enabling deno linting, then enabling unstable APIs. 

But make sure you have Deno installed as an extension first. Hope this works for you.

**Reply by carcinocron** [reply]
I've done all that and get the same error message

**Reply by justinekatebe** [reply]
try changing the url to "https://deno.land/std@0.177.0/http/server.ts" 🤔

**Reply by carcinocron** [reply]
it's not the URL, I'm getting this error for all deno URL imports:
```
error TS2307: Cannot find module 'https://esm.sh/@supabase/supabase-js@2' or its corresponding type declarations.

1 import { createClient } from 'https://esm.sh/@supabase/supabase-js@2'
```
and the code is question deploys and works on supabase servers

**Reply by justinekatebe** [reply]
I see... Do you have an "import_map.json" file in the .vscode folder (supabase/functions/.vscode)?

**Reply by carcinocron** [reply]
I don't have any files with a name similar to `import_map`.

**Reply by justinekatebe** [reply]
Ok, try following this supabase tutorial https://youtu.be/lFhU3L8VoSQ (Start from 03:03)

**Reply by carcinocron** [reply]
I figured it out, this is the command I was looking for:
```bash
deno lint --unstable
```
Video was still helpful.
